File: LeetCode/66. Plus One/Solution.cs
public class Solution
{
    public int[] PlusOne(int[] digits)
    {
        var carry = 1;
        var len = digits.Length;
        for (var i = len - 1; i >= 0; i--)
        {
            carry += digits[i];
            digits[i] = carry % 10;
            carry = carry / 10;
        }

        if (carry == 1)
        {
            var result = new int[len + 1];
            result[0] = 1;

            return result;
        }
        else return digits;
    }
}

File: LeetCode/66. Plus One/Solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1

        for i in range(len(digits) - 1, -1, -1):
            if carry == 0 : return digits
            if carry + digits[i] == 10:
                carry = 1
                digits[i] = 0
            else:
                digits[i] += 1
                carry = 0
        if carry == 1:
            return [1] + digits
        else: return digits
